utils/smartpaste/mattermost.tsx

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Options} from 'turndown';
+
+export const channelMentionsRule = {
+    filter: (node: Node): boolean => {
+        return node.nodeName === 'A' && (node as HTMLElement).className === 'mention-link' && Boolean((node as HTMLElement).getAttribute('data-channel-mention'));
+    },
+    replacement: (content: string, node: Node): string => {
+        const channelName = (node as HTMLElement).getAttribute('data-channel-mention');
+        if (channelName) {
+            return `~${channelName}`;
+        }
+        return '';
+    }
+};
+
+export const hashtagsRule = {
+    filter: (node: Node): boolean => {
+        return node.nodeName === 'A' && (node as HTMLElement).className === 'mention-link' && Boolean((node as HTMLElement).getAttribute('data-hashtag'));
+    },
+    replacement: (content: string, node: Node): string => {
+        return (node as HTMLElement).getAttribute('data-hashtag') || '';
+    }
+};
+
+export const filePreviewButtonRule = {
```

Comment:
What's this rule used for?

Reply:
This fix a problem related to coping and pasting markdown images from the existing posts. I added a comment in the code, to clarify that.